pallets/subtensor/src/macros/hooks.rs

```diff
@@ -127,4 +128,30 @@ mod hooks {
             Ok(())
         }
     }
+
+    impl<T: Config> Pallet<T> {
+        // This function is to clean up the old hotkey swap records
+        // It just clean up for one subnet at a time, according to the block number
+        fn clean_up_hotkey_swap_records(block_number: BlockNumberFor<T>) {
+            let hotkey_swap_on_subnet_interval = T::HotkeySwapOnSubnetInterval::get();
+            let block_number: u64 = TryInto::try_into(block_number)
+                .ok()
+                .expect("blockchain will not exceed 2^64 blocks; QED.");
+            let netuids = Self::get_all_subnet_netuids();
+
+            if let Some(slot) = block_number.checked_rem(hotkey_swap_on_subnet_interval) {
+                if slot < (u16::MAX as u64) && netuids.contains(&(slot as u16)) {
+                    for (coldkey, swap_block_number) in
+                        LastHotkeySwapOnNetuid::<T>::iter_prefix(slot as u16)
```

Comment:
Maybe it can cause some issue if there is a high number of hotkeys on the subnet, it could be safer to do this in on initialize and record the weights used?